src/browser/convert_options.js

```diff
@@ -195,87 +180,66 @@ function fetchLocalConfig(configUrl, successCallback, errorCallback) {
+
+        const minHeight = newOptions.minHeight;
+        const maxHeight = newOptions.maxHeight;
+        const defaultHeight = newOptions.defaultHeight;
+        if (defaultHeight < minHeight) {
+            newOptions.height = minHeight;
+        } else if (maxHeight !== -1 && defaultHeight > maxHeight) {
+            newOptions.height = maxHeight;
+        } else {
+            newOptions.height = defaultHeight;
+        }
 
-            let defaultWidth = newOptions.defaultWidth;
-            let minWidth = newOptions.minWidth;
-            let maxWidth = newOptions.maxWidth;
-            if (defaultWidth < minWidth) {
-                newOptions.width = minWidth;
-            } else if (maxWidth !== -1 && defaultWidth > maxWidth) {
-                newOptions.width = maxWidth;
-            } else {
-                newOptions.width = defaultWidth;
-            }
+        const defaultWidth = newOptions.defaultWidth;
+        const minWidth = newOptions.minWidth;
+        const maxWidth = newOptions.maxWidth;
+        if (defaultWidth < minWidth) {
+            newOptions.width = minWidth;
+        } else if (maxWidth !== -1 && defaultWidth > maxWidth) {
+            newOptions.width = maxWidth;
+        } else {
+            newOptions.width = defaultWidth;
+        }
 
-            newOptions.center = newOptions.defaultCentered;
-            if (!newOptions.center) {
-                newOptions.x = newOptions.defaultLeft;
-                newOptions.y = newOptions.defaultTop;
-            }
+        newOptions.center = newOptions.defaultCentered;
+        if (!newOptions.center) {
+            newOptions.x = newOptions.defaultLeft;
+            newOptions.y = newOptions.defaultTop;
         }
 
-        // Electron BrowserWindow options
         newOptions.enableLargerThanScreen = true;
         newOptions['enable-plugins'] = true;
         newOptions.webPreferences = {
             nodeIntegration: false,
-            plugins: newOptions.plugins
+            plugins: newOptions.plugins,
+            webSecurity: newOptions.webSecurity && !coreState.argo['disable-web-security']
         };
 
-        if (coreState.argo['disable-web-security'] || newOptions.webSecurity === false) {
-            newOptions.webPreferences.webSecurity = false;
+        // handle raw window.open; for internal use
+        if (options.rawWindowOpen !== undefined) {
+            newOptions.rawWindowOpen = options.rawWindowOpen;
         }
 
+        // used internally by notifications
         if (options.message !== undefined) {
             newOptions.message = options.message;
         }
 
-        if (options.customData !== undefined) {
-            newOptions.customData = options.customData;
-        }
-
-        if (options.permissions !== undefined) { // API policy
-            newOptions.permissions = options.permissions;
-        }
-
-        if (options.hasOwnProperty('preload')) {
-            newOptions.preload = options.preload;
-        }
```

Comment:
I don't fully trust `validate` function, as it won't update options with different types. Not sure we use different types for any of the options now, but maybe we will in the future and this will become a problem.
Also, arrays become problems too, as they are objects as well.